test/boost/types_test.cc

```diff
@@ -292,11 +292,14 @@ void test_timestamp_like_string_conversions(data_type timestamp_type) {
 
     // test time_stamps around year 0
     tp = db_clock::time_point(db_clock::duration(-62167219200000));
-    BOOST_REQUIRE_EQUAL(timestamp_type->to_string(timestamp_type->decompose(tp)), "0-01-01T00:00:00");
+    auto s = timestamp_type->to_string(timestamp_type->decompose(tp));
+    BOOST_REQUIRE(s == "0000-01-01T00:00:00" || s == "0-01-01T00:00:00");
     tp = db_clock::time_point(db_clock::duration(-62167219199211));
-    BOOST_REQUIRE_EQUAL(timestamp_type->to_string(timestamp_type->decompose(tp)), "0-01-01T00:00:00.789000");
+    s = timestamp_type->to_string(timestamp_type->decompose(tp));
+    BOOST_REQUIRE(s == "0000-01-01T00:00:00.789000" || s == "0-01-01T00:00:00.789000");
     tp = db_clock::time_point(db_clock::duration(-62167219200789));
-    BOOST_REQUIRE_EQUAL(timestamp_type->to_string(timestamp_type->decompose(tp)), "-1-12-31T23:59:59.211000");
+    s = timestamp_type->to_string(timestamp_type->decompose(tp));
+    BOOST_REQUIRE(s == "-0001-12-31T23:59:59.211000" || s == "-1-12-31T23:59:59.211000");
```

Comment:
Isn't this string something user-visible? If yes, is that OK that it looks differently at times?

Reply:
Other than this. On my node this line still fails:
```
test/boost/types_test.cc(302): fatal error: in "test_timestamp_string_conversions": critical check s == "-0001-12-31T23:59:59.211000" || s == "-1-12-31T23:59:59.211000" has failed
```
because this time `s == "-001-12-31T23:59:59.211000"` :roll_eyes:

Reply:
> Isn't this string something user-visible? If yes, is that OK that it looks differently at times?

AFAIK the printed format is used only for debugging / intrrnal purposes and is not part of the CQL interface. Not sure about the REST API, but at any rate, all variant should be good for parsing back into a time point.

> Other than this. On my node this line still fails:
> 
> ```
> test/boost/types_test.cc(302): fatal error: in "test_timestamp_string_conversions": critical check s == "-0001-12-31T23:59:59.211000" || s == "-1-12-31T23:59:59.211000" has failed
> ```
> 
> because this time `s == "-001-12-31T23:59:59.211000"` 🙄

width 4 by all means...
will allow that too :-/